services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeYhteystietoEntity.kt

```diff
@@ -41,4 +41,35 @@ class HankeYhteystietoEntity (
         @ManyToOne(fetch = FetchType.EAGER)
         @JoinColumn(name="hankeid")
         var hanke: HankeEntity? = null
-)
+) {
+
+    // Must consider both id and all non-audit fields for correct operations in certain collections
+    // Id can not be used as the only comparison, as one can have entities with null id (before they get saved).
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other !is HankeYhteystietoEntity) return false
+
+        if (sukunimi != other.sukunimi) return false
+        if (etunimi != other.etunimi) return false
+        if (email != other.email) return false
+        if (puhelinnumero != other.puhelinnumero) return false
+        if (organisaatioId != other.organisaatioId) return false
+        if (organisaatioNimi != other.organisaatioNimi) return false
+        if (osasto != other.osasto) return false
+
+        return true
```

Comment:
Id is the one which tells if there are the same.

Reply:
Between new entries, id is null on all of them. However, id comparison is still needed here (for updates), and contactType is missing (required). Coming...